### ISSUE TITLE
Can't create new line in messages

### ISSUE THREAD

**Issue by taha-ghadirian** [telegramdesktop/tdesktop#27957]
### Steps to reproduce

1. Open a chat
2. try to write down a text like:
```
line1


line 4

line 6
```

### Expected behaviour

New lines are created when pressing `Shift+Enter`, and the result should be like this:

```
line1
(empty line)
(empty line)
line 4
(empty line)
line 6
```

### Actual behaviour

Can't create multiple lines, the result will be like this:
```
line1
line 4
line 6
```

### Operating system

Arch Linux, 6.9.1, Hyprland WM on Wayland

### Version of Telegram Desktop

5.0.2

### Installation source

Static binary from official website

**Comment by Aokromes** [comment]
works for me on windows 10

https://i.imgur.com/ZleZGFq.png

**Comment by TsentsevitskyDmitry** [comment]
Similar behavior, but I can add at least one empty line.

### Expected behaviour
New lines are created when pressing Shift+Enter, then I can add one more text line, and the result should be like this:
```
line1
(empty line)
(empty line)
(empty line)
line2
```

### Actual behaviour
I can only add one empty line:
```
line1

line2
```

But if `line2` is already written, then I can add as many empty lines as I want between them.

### Operating system
Ubuntu 22.04 LTS, Xorg

### Version of Telegram Desktop
5.0.2

### Installation source
Static binary from official website

**Comment by saturn99** [comment]
on `debian` i have similar behavior
i can create only one new line with `ctrl + enter` or `shift + enter`

**Comment by Aokromes** [comment]
yeah it looks to be linux only bug.

**Comment by ilya-fedin** [comment]
By a quick look this looks like a new Qt 6.7.1 regression

**Comment by taha-ghadirian** [comment]
> By a quick look this looks like a new Qt 6.7.1 regression

Can you explain it more please?

**Comment by ilya-fedin** [comment]
I was wrong, it's https://github.com/desktop-app/lib_ui/commit/41d1fe9232c1d09b1ada01a0cc552e5c66d74b89 actually

**Comment by Viiprogrammer** [comment]
Same behavior, I think, after update to 5.0.2. Ubuntu 22.04.4 LTS x86_64. Can't add empty lines.

**Comment by myii** [comment]
Two (unpleasant) workarounds that can be used until this is fixed:

1. Put a character on the blank line (e.g. `.`), move to the beginning of the line and add as many newlines as you like.
2. Hit `<left-arrow>+<return>` repeatedly (to go back to the end of the last line and add another newline).

Both are shown below:

https://github.com/telegramdesktop/tdesktop/assets/10231489/e3c47274-3528-4218-8c5b-5dc33f5dfdf2

**Comment by taha-ghadirian** [comment]
The workaround that I'm currently using is to put half-space in text with unicode `U+2009`. 
You can also add it to your keyboard as a keybind for `shift+space`. 

However, that is not a perfect workaround when writing fast or writing work letters that are needs a lot of editing.

**Comment by hairycactus** [comment]
I can create 1 or more empty lines between lines containing text.

On the other hand ... *(not sure if below is by intentional design, or a bug)*
- All empty 1st line(s) & all empty last line(s) are automatically removed after the message is sent.
- There are situations where user might wish to have empty line(s) before & after the main text in the sent message.

Test Case 1:
```
Line1: <empty>
Line2
Line3: <empty>
```
... becomes:
```
Line2
```

Test Case 2:
```
Line1: <empty>
Line2: <empty>
Line3
Line4: <empty>
Line5
Line6: <empty>
Line7: <empty>
```
... becomes:
```
Line3
Line4: <empty>
Line5
```

### Screenshot:
![TelegramDesktop5 0 2-BugEmptyLinesRemoved-26May24](https://github.com/telegramdesktop/tdesktop/assets/9943769/05d5cf5a-4780-4479-9266-9b0a43279fb1)

This can also be observed in my post about another issue -- see Scenarios 3 & 4, plus accompanying screenshot & video demo:
- https://github.com/telegramdesktop/tdesktop/issues/27962#issuecomment-2132289437

### Tested Telegram Desktop Versions:
- v5.0.2 x64 Portable (25 May 2024: Github release): Affected
- v5.0.1 x64 Portable (04 May 2024: Github release): Affected

### OS:
- Windows 10 Build 18363 x64 Home

**Comment by ilya-fedin** [comment]
> * All empty 1st line(s) & all empty last line(s) are automatically removed after the message is sent.

This is intended

**Comment by hairycactus** [comment]
Below might related to the situation I posted in my [above comment](https://github.com/telegramdesktop/tdesktop/issues/27957#issuecomment-2132331504) ...

In Telegram Android's[^1] message input box:

#### Initial Attempt:
1. Type some text in Line 1.
2. Line(s) 2/3/etc: Leave empty, & leave cursor anywhere in empty line(s) here.

OR:
1. Line(s) 1/2/3/etc: Leave empty.
2. Type some text in Line 2. Leave cursor here in Line 2, or anywhere in preceding empty line(s).

Then ...
3. Keep focus in message input box.
4. Navigate to Android device's home screen, or go to another app (eg. to copy some text, with intention to paste into the empty line(s) within Telegram Android's message input box).
5. Return to Telegram Android.
6. Outcome: Empty Line(s) 2/3/etc in message input box has/have been removed, & cursor returns to end of text in Line 1.

#### Test Case 1:
```
Line1: Some text text text
Line2: <empty>
```

#### Test Case 2:
```
Line1: Some text text text
Line2: <empty>
Line3: <empty>
```
... both of above become:
```
Line1: Some text text text
```

#### Test Case 3:
```
Line1: <empty>
Line2: Some text text text
Line2: <empty>
```
... becomes:
```
Line2: Some text text text
```

#### Test Case 4:
```
Line1: <empty>
Line2: <empty>
Line3: Some text text text
Line4: <empty>
Line5: <empty>
```
... becomes:
```
Line3: Some text text text
```

All above outcomes do not occur during subsequent attempts (steps described below).

#### Subsequent Attempts:
7. Retain existing text in message input box, & repeat creating empty line(s) as described above.
8. Repeat steps 3 to 6.
9. Outcome: Empty line(s) is/are *not* removed, & cursor remains where it was last left -- ref: original input shown in Test Cases 1 to 4.


### Affected Telegram Android Versions:
- v10.12.0 (4710) arm64-v8a (02 May 2024: Google Play Store)
- v10.x.0: Several recent past versions (2024: Google Play Store)

### Android Version:
- v12

[^1] Note:
1. It is not occurring in Telegram Desktop so far, but only in Telegram Android.
2. Posting here for info, because there is no `Issues` section for user feedback in Telegram Android's [official Github repo](https://github.com/DrKLO/Telegram).

**Comment by ilya-fedin** [comment]
> 1. Posting here for info, because there is no `Issues` section for user feedback in Telegram Android's [official Github repo](https://github.com/DrKLO/Telegram). [↩](#user-content-fnref-1-73906b7c0de4540a3c2a9bb5b429ccd3)

No one here is interested in telegram android issues

**Comment by hairycactus** [comment]
> @ilya-fedin: No one here is interested in telegram android issues

Thanks for the info. Is the development for Telegram Desktop & Telegram Android separate ?

If both Desktop & Android builds rely on the same base (or maybe not ?), I'm just wondering why the [behaviour as described in my above comment](https://github.com/telegramdesktop/tdesktop/issues/27957#issuecomment-2132367454) started occurring only in Telegram Android sometime in 2024.

If Telegram Android's modified behaviour is by intention (unfortunately, no changelog details), thankfully Telegram Desktop has not gone down the same route as of now.

**Comment by ilya-fedin** [comment]
> Thanks for the info. Is the development for Telegram Desktop & Telegram Android separate ?

Yes. Compare the code in the repo of android application and this one. It's completely different. Each client is standalone.

**Comment by hairycactus** [comment]
> @ilya-fedin: Yes. Compare the code in the repo of android application and this one. It's completely different. Each client is standalone.

I see. Thanks for explaining. 👍 I'm a layperson who is not educated in programming.

Grateful that Telegram Desktop exists, because this is how I use Telegram most of the time. I almost never send messages in Telegram Android due to screen size & touch screen difficulties.

**Comment by michalko58** [comment]
> ### Steps to reproduce
> 
>     1. Open a chat
> 
>     2. try to write down a text like:
> 
> 
> ```
> line1
> 
> 
> line 4
> 
> line 6
> ```
> 
> ### Expected behaviour
> 
> New lines are created when pressing `Shift+Enter`, and the result should be like this:
> 
> ```
> line1
> (empty line)
> (empty line)
> line 4
> (empty line)
> line 6
> ```
> 
> ### Actual behaviour
> 
> Can't create multiple lines, the result will be like this:
> 
> ```
> line1
> line 4
> line 6
> ```
> 
> ### Operating system
> 
> Arch Linux, 6.9.1, Hyprland WM on Wayland
> ### Version of Telegram Desktop
> 
> 5.0.2
> ### Installation source
> 
> Static binary from official website
> ### Crash ID
> 
> _No response_
> ### Logs
> 
> _No response_

I can confirm this on Linux Mint 21.3, in previous version it worked.

**Comment by john-preston** [comment]
This will be fixed in 5.0.3 version later today.

**Comment by myii** [comment]
All working again in `5.0.3`, thanks.
